storage/postgres/subquery.go

```diff
@@ -0,0 +1,140 @@
+/*
+ *    Copyright 2018 The Service Manager Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package postgres
+
+import (
+	"fmt"
+	"github.com/Peripli/service-manager/pkg/query"
+	"reflect"
+)
+
+type subQuery struct {
+	tableName        string
+	referenceColumns string
+	dbTags           []tagType
+	sql              queryStringBuilder
+	queryParams      []interface{}
+
+	fieldCriteria []query.Criterion
+	orderByFields []orderRule
+	limit         string
+
+	whereClauseTreeCreator func([]query.Criterion) *whereClauseTree
+
+	err error
+}
+
+type selectSubQuery struct {
```

Comment:
whats the point of having subQuery and selectSubQuery ? subQuery is used only in select